cli/impl/src/test/java/io/helidon/build/cli/impl/CommandInvoker.java

```diff
@@ -285,6 +289,9 @@ private InvokerImpl(Builder builder) {
             groupId = builder.groupId == null ? config.defaultGroupId(substitutions) : builder.groupId;
             artifactId = config.artifactId(builder.artifactId, builder.projectName, substitutions);
             packageName = builder.packageName == null ? config.defaultPackageName(substitutions) : builder.packageName;
+            execScript = builder.execScript;
```

Comment:
You could use an enum like this:

```java
enum InvokerMode { CLASSPATH, EMBEDDED, EXECUTABLE }
```

Then use two builder attributes/methods:
```java
.embedded()
.executable(Path executable)
```

If not of these methods are used `InvokerMode.CLASSPATH` is used.
If `builder.executable != null` `InvokerMode.EXECUTABLE` is used
If `embedded == true` `InvokerMode.EMBEDDED is used`

The enum is not a must-have ; but it's easier to explain with it :)